### PR TITLE
Use Go modules (replaces glide)

### PR DESCRIPTION
**Pull request by hwh33**
Tests are failing for me, but I can't compare with master as I can't get `glide` to build the vendor folder.  Does anyone know if the tests ever worked?  It seems like they were already broken as there was a compilation issue unrelated to any dependencies.  Is this repo in use?

### CODE REVIEW

File: lantern_test.go [hunk 1]
```diff
@@ -25,6 +25,7 @@ func (c testUserConfig) AfterStart()              {}
 func (c testUserConfig) BandwidthUpdate(int, int) {}
 func (c testUserConfig) ConfigUpdate(bool)        {}
 func (c testUserConfig) ShowSurvey(survey string) {}
+func (c testUserConfig) SetStaging(bool)          {}
```
Comment (hwh33) [inline lantern_test.go:28]: Drive-by fix for a compilation issue.  This type needs to implement the UserConfig interface.  As that interface is implemented by this package, this could not be something broken by a dependency update.  This is why I think the tests were already broken.

File: lantern_test.go [hunk 2]
```diff
@@ -56,7 +57,7 @@ func testProxiedRequest(proxyAddr string, socks bool) error {
 		URL: &url.URL{
 			Scheme: "http",
 			Host:   "www.google.com",
-			Path:   "http://www.google.com/humans.txt",
+			Path:   "/humans.txt",
```
Comment (hwh33) [inline lantern_test.go:60]: Drive-by fix while I was trying to see if I could get the tests to work.  Unless there's something unique going on here, URL.Path refers to the sub-path under a host.

**Comment by oxtoacart** [comment]
This repo is not in use, closing PR.

**Comment by hwh33** [comment]
Sounds good
